src/main/java/com/sumindaa/demo/employeeservice/service/EmployeeService.java

```diff
@@ -3,21 +3,30 @@
 import com.sumindaa.demo.employeeservice.model.Employee;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
 @Service
 public class EmployeeService {
-    private List<Employee> employees = Arrays.asList(
+    private List<Employee> employees = new ArrayList(Arrays.asList(
             new Employee("suminda", "amara", "male"),
             new Employee("chaminda", "amara", "male"),
             new Employee("sethu", "amara", "female")
 
 
-    );
+    ));
 
 
-    public void updateEmployee(int id, Employee employee) {
+    public void updateEmployee(String id, Employee employee) {
+        for (int i = 0; i < employees.size(); i++) {
+            Employee t = employees.get(i);
+            if (t.getFirstName().equals(id)) {
+             employees.set(i, employee);
+             return;
+
+            }
+        }
```

Comment:
this is not the way to find an employee. find employee by id, not by first name

Reply:
Only First name, last name and gender have been added as properties of the employee. If the ID should be added, how do we add them. As auto generated or as manual?

Reply:
Good question, It should be auto-generated. Think about a solution. 
A special rule. A given ID can not be assigned to another employee